### DISCUSSION TITLE
How to determine center of inside edges h,w & x,y ?

### DISCUSSION THREAD

**Discussion by marshallapparalautomation** [ImageMagick/ImageMagick#6175] [Help]
### ImageMagick version

7.1.1-3

### Operating system, version and so on

Mac OSX Ventura

### Description

Im trying to figure out best method to use -canny to calculate the inside open area of an image without assuming the lines generated could be defined as a closed shape.

For example taking an image of a picture frame and calculating the inside edges / position of a rectangle. There are other images that won't have a perfect closed inside frame like an image of a water bottle. I would like to use the lines generated by -canny to eliminate usable area until an inside shape can be defined by height, width and x y position of the shape.

**Comment by fmw42** [comment]
Please provide input examples and some output images from those inputs. Also provide your exact command line(s).

Do the canny operation and then trim it and get the trim coordinates from %@. Then crop your input to the same coordinates.

`crop_vals=$(magick result_from _canny.suffix -format "%@" info:)`

**Reply by marshallapparalautomation** [reply]
how do I crop to the inside open area and get the hw & x,y of the shape compared to the original?

Here are a couple images of examples original and outfile of canny. The end goal is to be able to overlay an image to the dimension.


![cannyoutwaterbottle](https://user-images.githubusercontent.com/68432172/226135908-9ec2335e-ae60-4de8-880b-55a8e1a2fad4.png)
![inwaterbottle](https://user-images.githubusercontent.com/68432172/226135909-a7fd7400-9a32-4b53-afa7-a11aa77d4219.jpg)

![outfileflat](https://user-images.githubusercontent.com/68432172/226135791-b61b6b21-f059-48aa-8618-78f760527045.png)
![infileFlat](https://user-images.githubusercontent.com/68432172/226135792-f8b49f4a-5804-4485-bec4-9d4c49271b0e.jpg)
content.com/68432172/226135785-f40844ab-8da0-4649-8a77-e6081493fc27.jpg)
thubusercontent.com/68432172/226135710-0f4088cd-be23-45c6-8ec6-3a2cea340856.png)

**Comment by marshallapparalautomation** [comment]
@fmw42 how can I flood fill or create a frame to flood fill on the water bottle example inside the lines? You could try doing -kmeans to make the colors constant before doing the -canny.

**Comment by fmw42** [comment]
Try flood filling from the center. You may have to clean up the shine on the bottle before doing canny to get rid of the edge lines inside the bottle 

Can you explain further the result you want and why? I do not understand cropping inside an object that is empty.

**Reply by marshallapparalautomation** [reply]
@fmw42 Sorry to bug you here. Your web sites down and assuming your email isn't working? What's best method to ask you about one of your scripts? 'embroidery'. Is there an email that is good or would you prefer here?

**Reply by fmw42** [reply]
My email should be fine (fmw at alink dot net).

**Reply by marshallapparalautomation** [reply]
Making sure you got...
